tsdb/head_wal.go

```diff
@@ -74,6 +74,11 @@ func (h *Head) loadWAL(r *wlog.Reader, syms *labels.SymbolTable, multiRef map[ch
 		decodeErr, seriesCreationErr error
 	)
 
+	_, last, err := wlog.Segments(h.wal.Dir())
```

Comment:
nit: this is breaking the layers a bit - `r` is an abstract reader of one segment.
Maybe pass in `last` from the caller?

Reply:
done in https://github.com/prometheus/prometheus/pull/16060/commits/d55d468e8fb083ca44a783f837bc0d9c8e43440e